_posts/2021-09-28-signed-immutables-bug.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: post
+published: true
+title: 'Signed Immutables Bug'
+date: '2021-09-28'
+author: Solidity Team
+category: Security Alerts
+---
+
+On September 28th, 2021, the Solidity team discovered that
+for immutable variables of a signed integer type shorter than 256 bits,
+sign extension (cleanup) of its value is not always properly performed.
+
```

Comment:
```suggestion
The Solidity compiler is regularly fuzz tested using a mix of generational (Solidity programs generated based on language grammar) and mutational (mutated variants of existing Solidity programs) test creation engines. To the best of our knowledge, this bug was not found as part of the fuzzing effort because, on the one hand the generational component does not generate inline assembly as yet, and on the other hand, the mutational component is not efficient enough to synthesize the inline assembly component of the test program required to trigger the said bug. We plan to address this shortcoming by supporting generation of semantically correct inline assembly code snippets within Solidity programs.

```